src/pages/index.js

```diff
@@ -0,0 +1,304 @@
+
+  const deleteButton = cardElement.querySelector(".card__delete-btn");
+
+  cardLikeBtn.addEventListener("click", (evt) => handleLike(evt, data._id));
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-btn_liked");
+  }
+
+  function handleLike(evt, id) {
+    const isLiked = cardLikeBtn.classList.contains("card__like-btn_liked");
+    api
+      .changeLikeStatus(data._id, isLiked)
+      .then((res) => {
+        cardLikeBtn.classList.toggle("card__like-btn_liked", res.isLiked);
+      })
+      .catch(console.error);
+  }
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data);
+  });
+
+  cardNameEl.textContent = data.name;
+
+  cardImageEl.src = data.link;
+
+  cardImageEl.alt = data.name;
+
+  return cardElement;
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", closeModalEsc);
+  modal.removeEventListener("mousedown", closeOverlay);
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", closeModalEsc);
+  modal.addEventListener("mousedown", closeOverlay);
+}
+
+function closeOverlay(e) {
+  if (e.target.classList.contains("modal")) {
+    closeModal(e.target);
+  }
+}
+
+function closeModalEsc(e) {
+  if (e.key === "Escape") {
+    const modalOpened = document.querySelector(".modal_opened");
+    closeModal(modalOpened);
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleCardFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  const inputValue = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addNewCard(inputValue)
+    .then((res) => {
+      renderCard(res);
+      closeModal(cardModal);
+      evt.target.reset();
+      disabledButton(avatarSubmitBtn, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      submitBtn.textContent = "Save";
+    });
+}
+
+function handleAvatarFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      const avatarImage = document.getElementById("user-avatar");
```

Comment:
* It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine